src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController.swift

```diff
@@ -200,8 +204,51 @@ extension ExposureDetectionViewController {
 			checkButton.isEnabled = true
 		case .manual:
 			footerView.isHidden = false
-			checkButton.setTitle(AppStrings.ExposureDetection.buttonRefresh, for: .normal)
-			checkButton.isEnabled = riskProvider.manualExposureDetectionState == .possible
+
+			let nextRefresh = riskProvider.nextExposureDetectionDate()
+			let now = Date()
+
+			if nextRefresh > now, countdown == nil {
+				scheduleCountdownTimer(to: nextRefresh)
+			} else if let countdown = countdown, countdown.end != nextRefresh {
+				// Make sure to schedule new countdown if the next refresh time has changed.
+				scheduleCountdownTimer(to: nextRefresh)
+			} else if nextRefresh - 1 > now {
+				showCountdownButton(with: time)
+			} else {
+				showActiveRefreshButton()
+			}
 		}
 	}
+
+	private func scheduleCountdownTimer(to end: Date) {
+		countdown = CountdownTimer(countdownTo: end)
+		countdown?.delegate = self
+		countdown?.start()
+	}
+
+	private func showActiveRefreshButton() {
+		self.checkButton.setTitle(AppStrings.ExposureDetection.buttonRefresh, for: .normal)
+		self.checkButton.isEnabled = riskProvider.manualExposureDetectionState == .possible
```

Comment:
It seems odd to me that a method that is called `showActiveRefreshButton()` uses a boolean check to set a button to an enabled or disabled state. Does that make sense to you either?

Reply:
here i wanted to make sure that the button does not become clickable when the riskProvider does not explicitly allow updating. I add a comment to explain this defensive implementation.

Reply:
Considering the state the  HomeVC + all of its related classes are, this defensive approach is alright. Thanks for clarifying.